src/gui/guiChatConsole.cpp

```diff
@@ -773,6 +798,52 @@ bool GUIChatConsole::weblinkClick(s32 col, s32 row)
 	return false;
 }
 
+#ifdef USE_CURL
+// This is a function for CURL to write the page's html into a string
+size_t writeHtml(void* contents, size_t size, size_t nmemb, std::string* strptr) {
+    size_t totalSize = size * nmemb;
+    strptr->append((char*)contents, totalSize);
+    return totalSize;
+}
+#endif
+
+void GUIChatConsole::weblinkClickTitle(s32 col, s32 row){
+	std::ostringstream msg;
+	msg << " * ";
+#ifdef USE_CURL
+	std::string weblink, html, title;
+	weblink = getLinkAt(col, row);
+	CURL* curl = curl_easy_init();
+
+	if(weblink.empty() || !curl){
+		return;
+	}
+
+	curl_easy_setopt(curl, CURLOPT_URL, weblink.c_str());
+	curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writeHtml);
+	curl_easy_setopt(curl, CURLOPT_WRITEDATA, &html);
+	curl_easy_setopt(curl, CURLOPT_FOLLOWLOCATION, 1L); // Important for services like https://bit.ly or https://is.gd
+	curl_easy_setopt(curl, CURLOPT_TIMEOUT, 10L); // As we don't know if we are stuck in an infinite loop of redirects or if the specifiec server exists at ALL, 10 secs is max!
+	CURLcode res = curl_easy_perform(curl);
+	curl_easy_cleanup(curl);
+
+	if(res == CURLE_OPERATION_TIMEDOUT || res == CURLE_COULDNT_RESOLVE_HOST) {
+		msg  << gettext("Host does not exist or timeout for 3rd party webpages reached!");
+	} else {
+		if (res != CURLE_OK || html.find("<title>") == std::string::npos || html.find("</title>") == std::string::npos) {
```

Comment:
This is a pretty ugly hack. You're parsing HTML using `.find("<title>")` and `.find("</title>")`. This is not guaranteed to be reliable at all. I'm not sure how much / if this is an issue in practice, but I could very well imagine something like

```
<!-- <title>old title</title> -->
<title>new title</title>
```

Reply:
might be better if it searched for `og:title` (opengraph)

Reply:
This is just to ensure there are title tags in the html AT ALL

Reply:
> might be better if it searched for `og:title` (opengraph)

I didn't get what you mean, opengraph appears to be a libary for graphics, and adding a depdendency, even if optional, is not worth for parsing ONE value from html...

Reply:
> > might be better if it searched for `og:title` (opengraph)
> 
> Sorry, but I didn't get what you mean, opengraph appears to be a libary for graphics, and adding a depdendency, even if optional, is not worth for parsing ONE value from html...

something like `<meta property="og:title" content="The Rock" />` https://ogp.me/ - search `html opengraph tags` for more information

Reply:
I could just use the regex method with a string similar to this:

"<title>(.*?)</title>"